TESTS/mbed_hal/watchdog_reset/main.cpp

```diff
@@ -310,6 +354,12 @@ Case cases[] = {
 #endif
     Case("Watchdog started again", case_setup, test_restart_reset),
     Case("Kicking the Watchdog prevents reset", case_setup, test_kick_reset),
+    Case("timeout accuracy 20%", case_setup, test_timeout<200UL>),
```

Comment:
As we know, for some targets these tests will fail for sure if the watchdog clock is left uncalibrated. I think we should add `#if MBED_EXTENDED_TESTS` here, to skip these tests on the CI.